sdk/typescript/test/e2e/checkpoint.test.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, beforeAll } from 'vitest';
+// import { LocalTxnDataSerializer, RawSigner } from '../../src';
+import { setup, TestToolbox } from './utils/setup';
+
+describe('Checkpoints Reading API', () => {
+    let toolbox: TestToolbox;
+
+    beforeAll(async () => {
+      toolbox = await setup();
+    });
+
+    it('Get latest checkpoint sequence number', async () => {
+        const checkpointSequenceNumber = await toolbox.provider.getLatestCheckpointSequenceNumber();
+        expect(checkpointSequenceNumber).to.greaterThan(0);
+    });
+
+    it('Get checkpoint summary', async () => {
+        const resp = await toolbox.provider.getCheckpointSummary(0);
+        expect(resp.epoch).to.not.be.null;
+        expect(resp.sequence_number).to.not.be.null;
+        expect(resp.network_total_transactions).to.not.be.null;
+        expect(resp.content_digest).to.not.be.null;
+        expect(resp.epoch_rolling_gas_cost_summary).to.not.be.null;
+        expect(resp.timestamp_ms).to.not.be.null;
+    });
+
+    it('get checkpoint contents by digest', async () => {
+        const checkpoint_resp = await toolbox.provider.getCheckpointSummary(0);
+        const digest = checkpoint_resp.content_digest;
+        const resp = await toolbox.provider.getCheckpointSummaryByDigest(digest);
```

Comment:
The digest returned by this checkpoint summary doesn't seem to work when calling the test again.

Reply:
I suspected the issue is that `getCheckpointSummaryByDigest` requires a `CheckpointDigest` instead of a `CheckpointContentDigest`. Not sure how to get the `CheckpointDigest` though. @gegaowp Do you know since you added these?

Reply:
@666lcz is correct, I did not see a good way to get `CheckpointDigest` from checkpoint seq number b/c 
1. checkpoint [summary](https://github.com/MystenLabs/sui/blob/ba718140c643d4daa3a4226d511e9255bdac670c/crates/sui-types/src/messages_checkpoint.rs#L117) does not have `CheckpointDigest` field
2. checkpoint store has 2 maps, but no map from seq to CheckpointDigest
```
    certified_checkpoints: DBMap<CheckpointSequenceNumber, CertifiedCheckpointSummary>,
    checkpoint_by_digest: DBMap<CheckpointDigest, CertifiedCheckpointSummary>
```
other than testing, do you see a real use-case that needs this converting? will also ask andrey on slack and cc you both

Reply:
I will remove this test for now then, and just request a review of this ts sdk then.

Reply:
icymi, as mentioned [here]https://mysten-labs.slack.com/archives/C04FG4Q7YJ3/p1674684328522049
you can do summary.digest()

Reply:
Can we get the summary.digest() method from the TS client? All that is returned from the backend is the JSON right? I could modify the struct to call the method, and then pass the summary.digest() back?